File: backend/internal/models/Specialist.go
package models

import "time"

type Specialist struct {
	Id               int       `json:"id"`
	Name             string    `json:"name"`
	SecondName       string    `json:"second_name"`
	Email            string    `json:"email"`
	Description      string    `json:"description"`
	PhoneNr          string    `json:"phone_nr"`
	SpecializationId int       `json:"specialization_id"`
	CityId           int       `json:"city_id"`
	UserId           int       `json:"user_id"`
	CreatedAt        time.Time `json:"created_at"`
}

type SpecialistGeneralInfo struct {
	Id             int       `json:"id"`
	Name           string    `json:"name"`
	SecondName     string    `json:"second_name"`
	Specialization string    `json:"specialization"`
	City           string    `json:"city"`
	CreatedAt      time.Time `json:"created_at"`
	Rating         float32   `json:"rating"`
	Reviews        int       `json:"reviews"`
}

type SpecialistProfileInfo struct {
	Id             int       `json:"id"`
	Name           string    `json:"name"`
	SecondName     string    `json:"second_name"`
	Email          string    `json:"email"`
	Specialization string    `json:"specialization"`
	City           string    `json:"city"`
	PhoneNr        string    `json:"phone_nr"`
	Description    string    `json:"description"`
	CreatedAt      time.Time `json:"created_at"`
}

type SpecialistExtendedInfo struct {
	Info     SpecialistProfileInfo `json:"specialist"`
	Services []SpecialistService   `json:"services"`
	Reviews  []Review              `json:"reviews"`
}

type SpecialistBasicInfo struct {
	Id             int    `json:"id"`
	Name           string `json:"name"`
	SecondName     string `json:"second_name"`
	Specialization string `json:"specialization"`
	City           string `json:"city"`
}

File: frontend/src/models/Specialist.ts
import {SpecialistService} from "./SpecialistService";
import {Review} from "./Review";

export interface Specialist {
    id: number
    name: string
    second_name: string
    email: string
    description: string
    phone_nr: string
    specialization_id: number
    city_id: number
    user_id: number
    created_at: Date
}

export interface SpecialistGeneralInfo {
    id: number
    name: string
    second_name: string
    specialization: string
    city: string
    created_at: string
    rating: number
    reviews: number
}

export interface SpecialistProfileInfo {
    id: number
    name: string
    second_name: string
    email: string
    specialization: string
    city: string
    phone_nr: string
    description: string
    created_at: string
}

export interface SpecialistExtendedInfo {
    specialist: SpecialistProfileInfo
    services: SpecialistService[]
    reviews: Review[]
}
